src/main/java/controllers/ApplicationController.java

```diff
@@ -49,4 +49,14 @@ public Result sonar(Context context, AttackGameAction g) {
             return Results.badRequest();
         }
     }
+
+    public Result space(Context context, AttackGameAction g) {
+        Game game = g.getGame();
+        boolean result = game.attack(g.getActionRow(), g.getActionColumn());
+        if(result){
+            return Results.json().render(game);
+        }else{
+            return Results.badRequest();
+        }
+    }
```

Comment:
Leads to some repetition, but again- this makes sense in this case, and is hard to avoid here.